compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/codegen/CodeGenerator.java

```diff
@@ -634,7 +636,9 @@ public void visit(BLangLiteral literalExpr) {
                 break;
 
             case TypeTags.FLOAT:
-                double doubleVal = (Double) literalExpr.value;
+                double doubleVal = literalExpr.value instanceof String ?
+                        Double.parseDouble((String) literalExpr.value) :
```

Comment:
I think you shouldn't need to check this. This should always be double value as this will get parsed to double at TypeChecker